From 2bdf131ac2851e3c17ff93d6ffd7d019cfb0e670 Mon Sep 17 00:00:00 2001
From: "Hebbar, Gururaja" <gururaja.hebbar@ti.com>
Date: Fri, 13 Jan 2012 18:39:05 +0530
Subject: [PATCH 0702/1092] arm:omap:am335x: correct TPCC hwmod data

This patch adds missing TPTC hwmod data like addr space, 1 mpu irq, etc

Signed-off-by: Hebbar, Gururaja <gururaja.hebbar@ti.com>
---
 arch/arm/mach-omap2/omap_hwmod_33xx_data.c | 29 ++++++++++++++++++++++++++++-
 1 file changed, 28 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-omap2/omap_hwmod_33xx_data.c b/arch/arm/mach-omap2/omap_hwmod_33xx_data.c
index be358dc..59c0d30 100644
--- a/arch/arm/mach-omap2/omap_hwmod_33xx_data.c
+++ b/arch/arm/mach-omap2/omap_hwmod_33xx_data.c
@@ -68,6 +68,7 @@ static struct omap_hwmod am33xx_spi0_hwmod;
 static struct omap_hwmod am33xx_spi1_hwmod;
 static struct omap_hwmod am33xx_elm_hwmod;
 static struct omap_hwmod am33xx_adc_tsc_hwmod;
+static struct omap_hwmod am33xx_tpcc_hwmod;
 
 /*
  * Interconnects hwmod structures
@@ -2017,10 +2018,34 @@ static struct omap_hwmod_class am33xx_tpcc_hwmod_class = {
 
 /* tpcc */
 static struct omap_hwmod_irq_info am33xx_tpcc_irqs[] = {
-	{ .irq = AM33XX_IRQ_TPCC0_INT_PO0 },
+	{ .name	= "edma0", .irq = AM33XX_IRQ_TPCC0_INT_PO0 },
+	{ .name	= "edma0_err", .irq = AM33XX_IRQ_TPCC0_ERRINT_PO },
 	{ .irq = -1 },
 };
 
+/* TODO move this appropriate header. */
+#define AM33XX_TPCC_BASE		0x49000000
+
+static struct omap_hwmod_addr_space am33xx_tpcc_addr_space[] = {
+	{
+		.pa_start	= AM33XX_TPCC_BASE,
+		.pa_end		= AM33XX_TPCC_BASE + SZ_32K - 1,
+		.flags		= ADDR_TYPE_RT
+	},
+	{ }
+};
+
+static struct omap_hwmod_ocp_if am33xx_l3_slow__tpcc = {
+	.master		= &am33xx_l3slow_hwmod,
+	.slave		= &am33xx_tpcc_hwmod,
+	.addr		= am33xx_tpcc_addr_space,
+	.user		= OCP_USER_MPU,
+};
+
+static struct omap_hwmod_ocp_if *am33xx_tpcc_slaves[] = {
+	&am33xx_l3_slow__tpcc,
+};
+
 static struct omap_hwmod am33xx_tpcc_hwmod = {
 	.name		= "tpcc",
 	.class		= &am33xx_tpcc_hwmod_class,
@@ -2033,6 +2058,8 @@ static struct omap_hwmod am33xx_tpcc_hwmod = {
 			.modulemode	= MODULEMODE_SWCTRL,
 		},
 	},
+	.slaves		= am33xx_tpcc_slaves,
+	.slaves_cnt	= ARRAY_SIZE(am33xx_tpcc_slaves),
 };
 
 /* 'tptc' class */
-- 
1.7.11.2

